### PR TITLE
Python

### PR DESCRIPTION
**Pull request by jwisener**
Added support to generate the python client library for the scorm engine API. This should generate the objects that are needed for the python client.

Testing:
I ran this multiple times while developing the python client and created samples that pull data from both the java implementation (tomcat/mysql) and .net implementation (iis/sql server).

See attached files for the models that are generated from this.
[PythonModelOutput.zip](https://github.com/RusticiSoftware/json-schema-2-poco/files/424967/PythonModelOutput.zip)

Output from the actual executable. (that is generated for python). 
`Wrote ScormEngineClient_python\generated\RusticiSoftware\ScormContentPlayer\api\model\commentSchema.py
Wrote ScormEngineClient_python\generated\RusticiSoftware\ScormContentPlayer\api\model\configurationItem.py
Wrote ScormEngineClient_python\generated\RusticiSoftware\ScormContentPlayer\api\model\configListSchema.py
Wrote ScormEngineClient_python\generated\RusticiSoftware\ScormContentPlayer\api\model\courseSummarySchema.py
Wrote ScormEngineClient_python\generated\RusticiSoftware\ScormContentPlayer\api\model\courseListSchema.py
Wrote ScormEngineClient_python\generated\RusticiSoftware\ScormContentPlayer\api\model\idSchema.py
Wrote ScormEngineClient_python\generated\RusticiSoftware\ScormContentPlayer\api\model\importRequestSchema.py
Wrote ScormEngineClient_python\generated\RusticiSoftware\ScormContentPlayer\api\model\importResultSchema.py
Wrote ScormEngineClient_python\generated\RusticiSoftware\ScormContentPlayer\api\model\integerResultSchema.py
Wrote ScormEngineClient_python\generated\RusticiSoftware\ScormContentPlayer\api\model\launchLinkSchema.py
Wrote ScormEngineClient_python\generated\RusticiSoftware\ScormContentPlayer\api\model\linkSchema.py
Wrote ScormEngineClient_python\generated\RusticiSoftware\ScormContentPlayer\api\model\messageSchema.py
Wrote ScormEngineClient_python\generated\RusticiSoftware\ScormContentPlayer\api\model\minimalRegistrationSummarySchema.py
Wrote ScormEngineClient_python\generated\RusticiSoftware\ScormContentPlayer\api\model\scoreSchema.py
Wrote ScormEngineClient_python\generated\RusticiSoftware\ScormContentPlayer\api\model\objectiveSchema.py
Wrote ScormEngineClient_python\generated\RusticiSoftware\ScormContentPlayer\api\model\scoreSchema.py
Wrote ScormEngineClient_python\generated\RusticiSoftware\ScormContentPlayer\api\model\courseSummarySchema.py
Wrote ScormEngineClient_python\generated\RusticiSoftware\ScormContentPlayer\api\model\learnerSchema.py
Wrote ScormEngineClient_python\generated\RusticiSoftware\ScormContentPlayer\api\model\scoreSchema.py
Wrote ScormEngineClient_python\generated\RusticiSoftware\ScormContentPlayer\api\model\completionAmountSchema.py
Wrote ScormEngineClient_python\generated\RusticiSoftware\ScormContentPlayer\api\model\staticPropertiesSchema.py
Wrote ScormEngineClient_python\generated\RusticiSoftware\ScormContentPlayer\api\model\learnerPreferenceSchema.py
Wrote ScormEngineClient_python\generated\RusticiSoftware\ScormContentPlayer\api\model\runtimeInteractionSchema.py
Wrote ScormEngineClient_python\generated\RusticiSoftware\ScormContentPlayer\api\model\runtimeObjectiveSchema.py
Wrote ScormEngineClient_python\generated\RusticiSoftware\ScormContentPlayer\api\model\runtimeSchema.py
Wrote ScormEngineClient_python\generated\RusticiSoftware\ScormContentPlayer\api\model\activityResultSchema.py
Wrote ScormEngineClient_python\generated\RusticiSoftware\ScormContentPlayer\api\model\registrationSummarySchema.py
Wrote ScormEngineClient_python\generated\RusticiSoftware\ScormContentPlayer\api\model\registrationListSchema.py
Wrote ScormEngineClient_python\generated\RusticiSoftware\ScormContentPlayer\api\model\learnerSchema.py
Wrote ScormEngineClient_python\generated\RusticiSoftware\ScormContentPlayer\api\model\postBack.py
Wrote ScormEngineClient_python\generated\RusticiSoftware\ScormContentPlayer\api\model\registrationSchema.py
Wrote ScormEngineClient_python\generated\RusticiSoftware\ScormContentPlayer\api\model\settingSchema.py
Wrote ScormEngineClient_python\generated\RusticiSoftware\ScormContentPlayer\api\model\stringResultSchema.py
Wrote ScormEngineClient_python\generated\RusticiSoftware\ScormContentPlayer\api\model\titleSchema.py
Wrote ScormEngineClient_python\generated\RusticiSoftware\ScormContentPlayer\api\model\__init__.py
Wrote ScormEngineClient_python\generated\RusticiSoftware\__init__.py
Wrote ScormEngineClient_python\generated\RusticiSoftware\ScormContentPlayer\__init__.py
Wrote ScormEngineClient_python\generated\RusticiSoftware\ScormContentPlayer\api\__init__.py`

### CODE REVIEW

File: Source/Cvent.SchemaToPoco.Core/CodeToLanguage/PythonCodeGenerator.cs [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+        }
+
+
+        private void GeneratePythonFileStart()
+        {
+            
+            //_output.Indent = 1;
+            _output.WriteLine("\"\"\"");
+            _output.WriteLine(@" <auto-generated>");
+            _output.WriteLine(@"     This code was generated by a tool.");
```
Comment (bscSCORM) [inline Source/Cvent.SchemaToPoco.Core/CodeToLanguage/PythonCodeGenerator.cs]: best to name the tool here (jsonSchema2Poco)

File: Source/Cvent.SchemaToPoco.Core/CodeToLanguage/PythonCodeGenerator.cs [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+                else
+                {
+                    if (member is CodeMemberField)
+                    {
+
+                        var propertyName = member.Name.Replace(" { get; set; } //", string.Empty);
+
+                        //getter
+                        _output.WriteLine("    self.{0} = {0}", StringUtils.LowerFirst(propertyName));
+                        //_output.WriteLine("@property");
```
Comment (bscSCORM) [inline Source/Cvent.SchemaToPoco.Core/CodeToLanguage/PythonCodeGenerator.cs]: we should probably get rid of this commented out code

File: Source/Cvent.SchemaToPoco.Core/JsonSchemaToPoco.cs [hunk 3]
```diff
@@ -152,6 +161,39 @@ private void Generate()
...
+                sep = ",\n" + new String(' ', 11);
+            }
+            IoUtils.GenerateFile(allVar + "]", initpyFile);
+
+            //ignore last directory we just created that file
+            //before this code block.
+            var dirName = "";
+            for (int i = 0; i < directories.Length - 1; i++)
+            {
+                dirName = dirName + "\\" + directories[i];
```
Comment (bscSCORM) [inline Source/Cvent.SchemaToPoco.Core/JsonSchemaToPoco.cs]: `Path.Combine(dirName, directories[i])`
(so this works properly on linux / mono too)

Comment (bscSCORM) [inline Source/Cvent.SchemaToPoco.Core/JsonSchemaToPoco.cs]: please go through and look for other `\\` to change to `Path.Combine` or `Path.PathSeparator`

**Comment by bscSCORM** [comment]
@jwisener it's probably worth fixing the path related comment at least. I'd like to see a sample of the output in the PR as well.

**Comment by jwisener** [comment]
Okay, I think I've covered your feedback and committed the needed changes.

**Comment by bscSCORM** [comment]
👍 @kpatmor and note, `__pycache__` stuff in the zip was not generated by this tool, it just got created during testing then included in the zip.

**Pull request merged**